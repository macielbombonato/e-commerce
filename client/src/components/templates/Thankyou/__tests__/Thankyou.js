import createTestContext from 'react-cosmos-test/enzyme';

import fixtureOnBoleto from '../__fixtures__/onBoleto';
import fixtureOnCreditCard from '../__fixtures__/onCreditCard';

describe('#Thankyou template', () => {
  describe('#onBoleto', () => {
    const { mount, getWrapper } = createTestContext({ fixture: fixtureOnBoleto });

    beforeEach(mount);

    it('renders onBoleto template', () => {
      expect(getWrapper()).toMatchSnapshot();
    });
  });

  describe('#onCreditCard', () => {
    const { mount, getWrapper } = createTestContext({ fixture: fixtureOnCreditCard });

    beforeEach(mount);

    it('renders onCreditCard template', () => {
      expect(getWrapper()).toMatchSnapshot();
    });
  });
});
