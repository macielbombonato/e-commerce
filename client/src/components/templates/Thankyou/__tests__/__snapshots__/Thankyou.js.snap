// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Thankyou template #onBoleto renders onBoleto template 1`] = `
<Thankyou
  checkout={
    Object {
      "status": Object {
        "boleto_url": "http://test.com",
        "payment_method": "boleto",
      },
    }
  }
>
  <Container
    tag="div"
  >
    <div
      className="container"
    >
      <div
        className="mb-4"
      >
        Obrigado por comprar em nossa loja.
      </div>
      <div>
        <a
          className="mb-4"
          href="http://test.com"
        >
          Clique aqui para pagar seu boleto
        </a>
      </div>
      <Link
        replace={false}
        to="/"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Página Principal
        </a>
      </Link>
    </div>
  </Container>
</Thankyou>
`;

exports[`#Thankyou template #onCreditCard renders onCreditCard template 1`] = `
<Thankyou
  checkout={
    Object {
      "status": Object {
        "payment_method": "credit_card",
      },
    }
  }
>
  <Container
    tag="div"
  >
    <div
      className="container"
    >
      <div
        className="mb-4"
      >
        Obrigado por comprar em nossa loja.
      </div>
      <Link
        replace={false}
        to="/"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Página Principal
        </a>
      </Link>
    </div>
  </Container>
</Thankyou>
`;
